### PR TITLE
Add routing config support to Sagemaker Endpoint Configuration

### PR DESCRIPTION
**Pull request by curtisallen**
### Description

Recently [AWS added support](https://aws.amazon.com/blogs/machine-learning/minimize-real-time-inference-latency-by-using-amazon-sagemaker-routing-strategies/) for routing strategies in Sagemaker Endpoints. This PR adds `routing_config` support to Sagemaker Config endpoints.

### CODE REVIEW

File: website/docs/r/sagemaker_endpoint_configuration.html.markdown [hunk 1]
```diff
@@ -64,6 +64,10 @@ This resource supports the following arguments:
 * `destination_s3_uri` - (Required) The Amazon S3 bucket to send the core dump to.
 * `kms_key_id` - (Required) The Amazon Web Services Key Management Service (Amazon Web Services KMS) key that SageMaker uses to encrypt the core dump data at rest using Amazon S3 server-side encryption.
 
+#### routing_config
```
Comment (DrFaust92) [inline website/docs/r/sagemaker_endpoint_configuration.html.markdown:67]: missing from top production_variants block

File: internal/service/sagemaker/endpoint_configuration_test.go [hunk 2]
```diff
@@ -775,6 +803,9 @@ resource "aws_sagemaker_endpoint_configuration" "test" {
     initial_instance_count = 2
     instance_type          = "ml.t2.medium"
     initial_variant_weight = 1
+    routing_config {
```
Comment (DrFaust92) [inline internal/service/sagemaker/endpoint_configuration_test.go:806]: I think the dedicated test is good enough, also this is not checked anywhere so lets revert this

File: internal/service/sagemaker/endpoint_configuration.go [hunk 3]
```diff
@@ -916,6 +950,22 @@ func expandEndpointConfigNotificationConfig(configured []interface{}) *sagemaker
 	return c
 }
 
+func expandRoutingConfig(configured []interface{}) *sagemaker.ProductionVariantRoutingConfig {
```
Comment (DrFaust92) [inline internal/service/sagemaker/endpoint_configuration.go:953]: Missing the flatten for the read

File: internal/service/sagemaker/endpoint_configuration.go [hunk 4]
```diff
@@ -448,6 +463,21 @@ func ResourceEndpointConfiguration() *schema.Resource {
...
+							Type:     schema.TypeList,
+							Optional: true,
+							ForceNew: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"routing_strategy": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ForceNew:     true,
+										ValidateFunc: validation.StringMatch(regexache.MustCompile(`^(LEAST_OUTSTANDING_REQUESTS|RANDOM)`), ""),
```
Comment (DrFaust92) [inline internal/service/sagemaker/endpoint_configuration.go:476]: Lets use string in list instead and check if API maybe has this as a list already (if not still better to check it as a static list)

Comment (DrFaust92) [reply internal/service/sagemaker/endpoint_configuration.go:476]: for example:

							ValidateFunc: validation.StringInSlice(sagemaker.ProductionVariantAcceleratorType_Values(), false),

**Review by DrFaust92 — CHANGES_REQUESTED** [review]
Good start, some minor changes

File: internal/service/sagemaker/endpoint_configuration_test.go (hunk 2 above)
Comment (curtisallen) [reply internal/service/sagemaker/endpoint_configuration_test.go:806]: oh right, will do, forgot I committed this bit

File: internal/service/sagemaker/endpoint_configuration.go (hunk 4 above)
Comment (curtisallen) [reply internal/service/sagemaker/endpoint_configuration.go:476]: oh thanks! I was looking all over the validation package and missed this, exactly what I wanted.

**Comment by curtisallen** [comment]
thanks for the speedy review @DrFaust92 , please take another look 🙇🏾

**Comment by curtisallen** [comment]
Happy new year @DrFaust92 gentle ping on this one 🙇🏾

**Review by DrFaust92 — APPROVED** [review]
LGTM

**Comment by curtisallen** [comment]
can we get a merge @DrFaust92 ?

**Review by jar-b — APPROVED** [review]
LGTM 🎉 

```console
% make testacc PKG=sagemaker TESTS=TestAccSageMakerEndpointConfiguration_
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/sagemaker/... -v -count 1 -parallel 20 -run='TestAccSageMakerEndpointConfiguration_'  -timeout 360m

--- PASS: TestAccSageMakerEndpointConfiguration_ProductionVariants_variantNameGenerated (29.79s)
=== CONT  TestAccSageMakerEndpointConfiguration_ProductionVariants_initialVariantWeight
--- PASS: TestAccSageMakerEndpointConfiguration_namePrefix (29.82s)
--- PASS: TestAccSageMakerEndpointConfiguration_async_includeInference (33.69s)
--- PASS: TestAccSageMakerEndpointConfiguration_Async_client_failurePath (36.06s)
=== NAME  TestAccSageMakerEndpointConfiguration_upgradeToEnableSSMAccess
    testing_new.go:80: Error retrieving state, there may be dangling resources: exit status 1
        Failed to marshal state to json: unsupported attribute "override_json"
--- FAIL: TestAccSageMakerEndpointConfiguration_upgradeToEnableSSMAccess (38.12s)
--- PASS: TestAccSageMakerEndpointConfiguration_Async_notif (39.89s)
--- PASS: TestAccSageMakerEndpointConfiguration_async_kms (43.91s)
--- PASS: TestAccSageMakerEndpointConfiguration_nameGenerated (47.65s)
--- PASS: TestAccSageMakerEndpointConfiguration_tags (48.80s)
--- PASS: TestAccSageMakerEndpointConfiguration_ProductionVariants_routing (51.40s)
--- PASS: TestAccSageMakerEndpointConfiguration_ProductionVariants_serverless (52.32s)
--- PASS: TestAccSageMakerEndpointConfiguration_ProductionVariants_initialVariantWeight (23.73s)
--- PASS: TestAccSageMakerEndpointConfiguration_async (67.22s)
--- PASS: TestAccSageMakerEndpointConfiguration_dataCapture (68.80s)
--- PASS: TestAccSageMakerEndpointConfiguration_Async_client (69.09s)
--- PASS: TestAccSageMakerEndpointConfiguration_kmsKeyID (69.29s)
--- PASS: TestAccSageMakerEndpointConfiguration_ProductionVariants_acceleratorType (70.69s)
--- PASS: TestAccSageMakerEndpointConfiguration_ProductionVariants_serverlessProvisionedConcurrency (76.68s)
--- PASS: TestAccSageMakerEndpointConfiguration_shadowProductionVariants (78.64s)
--- PASS: TestAccSageMakerEndpointConfiguration_basic (79.49s)
--- PASS: TestAccSageMakerEndpointConfiguration_disappears (81.61s)
FAIL
FAIL    github.com/hashicorp/terraform-provider-aws/internal/service/sagemaker  88.607s
```

Test failure is unrelated to this change.

**Comment by jar-b** [comment]
Thanks for your contribution, @curtisallen! 🎸

**Pull request merged**